logging/jsonlog/jsonlog.go

```diff
@@ -0,0 +1,327 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package jsonlog provides a Logger that logs structured JSON to Stderr by
+// default. When used on the various Cloud Compute environments (Cloud Run,
+// Cloud Functions, GKE, etc.) these JSON messages will be parsed by the Cloud
+// Logging agent and transformed into a message format that mirrors that of the
+// Cloud Logging API.
+package jsonlog
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"cloud.google.com/go/logging"
+	"cloud.google.com/go/logging/internal"
+	logtypepb "google.golang.org/genproto/googleapis/logging/type"
+	logpb "google.golang.org/genproto/googleapis/logging/v2"
+	"google.golang.org/protobuf/encoding/protojson"
+)
+
+// NewLogger creates a Logger that logs structured JSON to Stderr. The value of
+// parent must be in the format of:
+//    projects/PROJECT_ID
+//    folders/FOLDER_ID
+//    billingAccounts/ACCOUNT_ID
+//    organizations/ORG_ID
+func NewLogger(parent string, opts ...LoggerOption) (*Logger, error) {
```

Comment:
Can it just be called `Logger` for consistency? Or does the name need to be different from the `Logger` fn in logging.go?

Reply:
I think it should be different. Logger() implies just a logger is returned which is true for cloud.google.com/go/logging.Logger. That is not true here as we preform some light validation which can result in an error being returned. For the other Logger this same type of validation happens at the same time as connections dailing in NewClient I believe.

Because an error is also returned I think the `New` semantic works well here.